packages/library-base/src/commonMain/kotlin/io/realm/kotlin/Realm.kt

```diff
@@ -100,6 +101,28 @@ public interface Realm : TypedRealm {
             if (!fileExists(configuration.path)) return
             RealmInterop.realm_delete_files(configuration.path)
         }
+
+        /**
+         * Compacts the Realm file defined by the given configuration. Compaction can only succeed
+         * if all references to the Realm file has been closed.
+         *
+         * This method is not available on Windows (JVM), and will throw an [NotImplementedError]
+         * there.
+         *
+         * @param configuration configuration for the Realm to compact.
+         * @return `true` if compaction succeeded, `false` if not.
+         */
+        public fun compactRealm(configuration: Configuration): Boolean {
```

Comment:
That is actually a good point 👍

Reply:
It could make sense to raise awareness by making it a suspendable method,  but you can always turn any method info a suspending one with 
```
async(context) { compactRealm(...) }.await()
```
like going the opposite with `runBlocking`.  Have we followed the same reasoning for making the methods suspendable throughout the API 🤔